src/sources/prometheus/parser.rs

```diff
@@ -412,63 +119,37 @@ pub fn parse(packet: &str) -> Result<Vec<Metric>, ParserError> {
                     result.push(hist);
                 }
             }
-            ParserType::Summary => {
+            GroupKind::Summary(vec) => {
                 let mut aggregates = IndexMap::new();
 
-                for line in group {
-                    let metric = parse_metric(&line)?;
-                    let mut tags = metric.tags;
-                    let bucket = tags.remove("quantile");
-
-                    let (name, suffix) =
-                        if metric.name.ends_with("_sum") || metric.name.ends_with("_count") {
-                            let v: Vec<_> = metric.name.rsplitn(2, '_').collect();
-                            (v[1], v[0])
-                        } else {
-                            (&metric.name[..], "")
-                        };
-
-                    let mut id: Vec<_> = tags.iter().collect();
-                    id.sort();
-                    let group_key = format!("{:?}", id);
-
-                    let aggregate = aggregates.entry(group_key).or_insert(ParserAggregate {
-                        name: name.to_owned(),
+                for metric in vec {
+                    let tags = metric.get_labels().clone();
+                    let aggregate = aggregates.entry(tags.clone()).or_insert(ParserAggregate {
+                        name: group.name.clone(),
                         bounds: Vec::new(),
                         values: Vec::new(),
                         count: 0,
                         sum: 0.0,
                         tags,
                     });
 
-                    match suffix {
-                        "" => {
-                            if let Some(b) = bucket {
-                                aggregate.bounds.push(parse_value(&b)?);
-                                aggregate.values.push(metric.value);
-                            } else {
-                                return Err(ParserError::Malformed {
-                                    s: "expected \"quantile\" tag in summary bucket",
-                                    context: line,
-                                });
-                            }
-                        }
-                        "sum" => {
-                            aggregate.sum = metric.value;
+                    match metric {
+                        SummaryMetric::Count { value, .. } => {
+                            aggregate.count = value as u32; // TODO: check
```

Comment:
Does this still need checking?

Reply:
@bruceg I'm not sure, negative values here will be rounded to 0. Should we report an error for this?

Reply:
Probably, yes. Should at least output a warning.

Reply:
I added `ExpectNonNegativeValue`.